src/esm_parser/esm_parser.py

```diff
@@ -1881,6 +1882,7 @@ def determine_regex_list_match(test_str, regex_list):
     return any(result)
 
 
+@keep_id_provenance
```

Comment:
If I see it correctly, you add this decorator to every possible `recursive_run_function` function, yes?

I prefer the decorator, since it's right next to the function definition, but you could also do in `recursive_run_function`:

```python
def recursive_run_function(..., func, ...):
    # ... some other code
    func = keep_id_provenance(func)
    # ... rest of body
```

Reply:
I thought about that but there is the recursive function `list_to_multikey` that won't work like that because it actually mutates and transforms the content of a dictionary (it's or looping syntax). So for now I thought is better to have the decorator for the more simple recursive functions.